README.md

```diff
@@ -0,0 +1,434 @@
+  getUser,
+  refetch
+);
+
+declare function dispatchError(e: Error): void;
+declare function setCurrentUser(e: User): void;
+
+// feeding your query to `observe` will give you an observable on the query
+// N.B. until now no fetch is yet attempted, avenger will wait until the first subscription is issued
+const observable: Observable<QueryResult<Error, User>> = observe(userQuery);
+
+// after this the fetch function triggers
+observable.subscribe(dispatchError, setCurrentUser);
+
+// alternatively you can call `run` on your query and it will return a TaskEither<Error, User>
+// you can then use it imperatively
+const task: TaskEither<Error, User> = userQuery.run(1);
+const result: Either<Error, User> = await task.run();
+```
+
+altough the `run` method is available to check a query result imperatively, it is highly suggested the use of the `observe` utility in order to be notified in real time of when data changes. Either way, whenever you ask for a query result you will end up with an object with the [**`QueryResult`**](#QueryResult) signature that conveniently lets you fold to decide the best way to handle the result.
+
+
+the flow when you subscribe / call run on a cached value is relatively simple:
+!["cached flow"](docs/CachedValue.svg)
+
+
+
+when you call run / subscribe on a combination `query`/`inputs` that was never retrieved before (or whose last try to retrive it ended up with a `Failure`), avenger will try to run the `Fetch` function resulting in a more complex flow:
+!["cached flow"](docs/UncachedOrErrorValue.svg)
+
+## composing queries
+you can build bigger queries from smaller ones in two ways:
+
+- by composing them with [**`compose`**](#compose): when you need your queries to be sequentially run with the results of one feeding the other, you can use `compose`:
+
+```ts
+import { compose } from 'avenger/lib/Query';
+
+type UserPreferences = { color: string };
+
+// note that the two ends of the composed functions must have compatible types
+declare function getUser(userId: number): TaskEither<Error, User>;
+declare function getUserPreferences(
+  user: User
+): TaskEither<Error, UserPreferences>;
+
+const userQuery: ObservableQuery<number, Error, User> = queryShallow(
+  getUser,
+  refetch
+);
+
+const preferencesQuery: ObservableQuery<
+  User,
+  Error,
+  UserPreferences
+> = queryShallow(getUserPreferences, refetch);
+
+const composition: Composition<number, Error, UserPreferences> = compose(
+  userQuery,
+  preferencesQuery
+);
+```
+
+- by grouping them with [**`product`**](#product): when you don't need to run the queries sequentially but would like to conveniently group them and treat them as if they where one you can use `product`:
+
+```ts
+const grouped = product({ myQuery, myQuery2 });
+```
+
+# commands
+Since now we only described how fetching of data is performed, when you need to update / insert data remotely you can make use of [**`command`**](#command):
```

Comment:
```suggestion
Since now we only described how fetching of data is performed. When you need to update / insert data remotely you can make use of [**`command`**](#command):
```